fleetspeak/src/client/watchdog/watchdog.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2018 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package watchdog implements a simple watchdog timer which dumps a trace of
+// all goroutines to a file, and then ends the program, if the timer reaches its
+// limit.
+package watchdog
+
+import (
+	"io/ioutil"
+	"runtime/pprof"
+	"time"
+
+	log "github.com/golang/glog"
+)
+
+type Watchdog struct {
+	dir      string
+	prefix   string
+	duration time.Duration
+	reset    chan struct{}
+	noExit   bool
```

Comment:
I don't like this name, I think mainly because it's hard to parse due to the no... prefix. What do you think about exitAfterDump or similar?

Reply:
Renamed skipExit, added a comment.